libibnetdisc/CMakeLists.txt

```diff
@@ -20,5 +20,6 @@ rdma_pkg_config("ibnetdisc" "libibumad libibmad" "")
 rdma_test_executable(testleaks tests/testleaks.c)
 target_link_libraries(testleaks LINK_PRIVATE
   ibmad
+  ibumad
```

Comment:
This isn't how things are supposed to work, ibmad shuld be self contained

target_link_libraries(ibmad LINK_PRIVATE
  ibumad
  )

I assume changing the cmake version also changed how some of this stuff works?

Reply:
Yes, for some reason SuSE 15.0 didn't build without this change.
See first commit for an error.

Reply:
ok, I found the reason to missing -rpath error.
It was caused by CMP0022
https://cmake.org/cmake/help/latest/policy/CMP0022.html
I tried as a workaround the following change
cmake_policy(SET CMP0022 OLD)

and suse built correctly.

Reply:
That makes alot more sense

Reply:
But should we make the update described in that policy document, whatever it is?

Reply:
Yes, we need, the cmake_policy() was simplest way to check that we are hitting this specific policy change.